src/plugins/oauth2/openIdConfiguration.ts

```diff
@@ -43,30 +80,42 @@ export function getOpenIdConfiguration(
 ): models.OpenIdConfiguration {
   const config: models.OpenIdConfiguration = {
     variablePrefix: variablePrefix || DefaultOAuthVariablePrefix,
-    authorizationEndpoint: getVariable(variables, variablePrefix, 'authorizationEndpoint'),
-    deviceCodeEndpoint: getVariable(variables, variablePrefix, 'deviceCodeEndpoint'),
-    tokenEndpoint: getVariable(variables, variablePrefix, 'tokenEndpoint'),
-    clientId: getVariable(variables, variablePrefix, 'clientId'),
-    clientSecret: getVariable(variables, variablePrefix, 'clientSecret'),
-    responseType: getVariable(variables, variablePrefix, 'responseType'),
-    responseMode: getVariable(variables, variablePrefix, 'responseMode'),
-    audience: getVariable(variables, variablePrefix, 'audience'),
-    scope: getVariable(variables, variablePrefix, 'scope'),
-    resource: getVariable(variables, variablePrefix, 'resource'),
-    username: getVariable(variables, variablePrefix, 'username'),
-    password: getVariable(variables, variablePrefix, 'password'),
-    subjectIssuer: getVariable(variables, variablePrefix, 'subjectIssuer'),
+    authorizationEndpoint: getString(variables, variablePrefix, 'authorizationEndpoint'),
+    deviceCodeEndpoint: getString(variables, variablePrefix, 'deviceCodeEndpoint'),
+    tokenEndpoint: getString(variables, variablePrefix, 'tokenEndpoint'),
+    clientId: getString(variables, variablePrefix, 'clientId'),
+    clientSecret: getString(variables, variablePrefix, 'clientSecret'),
+    responseType: getString(variables, variablePrefix, 'responseType'),
+    responseMode: getString(variables, variablePrefix, 'responseMode'),
+    audience: getString(variables, variablePrefix, 'audience'),
+    scope: getString(variables, variablePrefix, 'scope'),
+    resource: getString(variables, variablePrefix, 'resource'),
+    username: getString(variables, variablePrefix, 'username'),
+    password: getString(variables, variablePrefix, 'password'),
+    subjectIssuer: getString(variables, variablePrefix, 'subjectIssuer'),
     redirectUri: getUrl(variables, variablePrefix, 'redirectUri', DEFAULT_CALLBACK_URI),
-    keepAlive: ['true', '1', true].indexOf(getVariable(variables, variablePrefix, 'keepAlive')) < 0,
-    useAuthorizationHeader:
-      ['false', '0', false].indexOf(getVariable(variables, variablePrefix, 'useAuthorizationHeader')) < 0,
-    useDeviceCodeClientSecret:
-      ['true', '1', true].indexOf(getVariable(variables, variablePrefix, 'useDeviceCodeClientSecret')) >= 0,
-    usePkce: ['true', '1', true].indexOf(getVariable(variables, variablePrefix, 'usePkce')) >= 0,
+    keepAlive: getBooleanLike(variables, variablePrefix, 'keepAlive', true),
```

Comment:
Thanks for refactoring the lines as well. They were really not pretty.

Reply:
I think that was actually husky doing it's thing 😂